src/data/binary_sensor.ts

```diff
@@ -0,0 +1,154 @@
+
+// On means light detected, Off means no light
+export const DEVICE_CLASS_LIGHT = "light";
+
+// On means open (unlocked), Off means closed (locked)
+export const DEVICE_CLASS_LOCK = "lock";
+
+// On means wet, Off means dry
+export const DEVICE_CLASS_MOISTURE = "moisture";
+
+// On means motion detected, Off means no motion (clear)
+export const DEVICE_CLASS_MOTION = "motion";
+
+// On means moving, Off means not moving (stopped)
+export const DEVICE_CLASS_MOVING = "moving";
+
+// On means occupied, Off means not occupied (clear)
+export const DEVICE_CLASS_OCCUPANCY = "occupancy";
+
+// On means open, Off means closed
+export const DEVICE_CLASS_OPENING = "opening";
+
+// On means plugged in, Off means unplugged
+export const DEVICE_CLASS_PLUG = "plug";
+
+// On means power detected, Off means no power
+export const DEVICE_CLASS_POWER = "power";
+
+// On means home, Off means away
+export const DEVICE_CLASS_PRESENCE = "presence";
+
+// On means problem detected, Off means no problem (OK)
+export const DEVICE_CLASS_PROBLEM = "problem";
+
+// On means unsafe, Off means safe
+export const DEVICE_CLASS_SAFETY = "safety";
+
+// On means smoke detected, Off means no smoke (clear)
+export const DEVICE_CLASS_SMOKE = "smoke";
+
+// On means sound detected, Off means no sound (clear)
+export const DEVICE_CLASS_SOUND = "sound";
+
+// On means vibration detected, Off means no vibration
+export const DEVICE_CLASS_VIBRATION = "vibration";
+
+// On means open, Off means closed
+export const DEVICE_CLASS_WINDOW = "window";
+
+export const DEVICE_CLASSES = [
+  DEVICE_CLASS_BATTERY,
+  DEVICE_CLASS_BATTERY_CHARGING,
+  DEVICE_CLASS_COLD,
+  DEVICE_CLASS_CONNECTIVITY,
+  DEVICE_CLASS_DOOR,
+  DEVICE_CLASS_GARAGE_DOOR,
+  DEVICE_CLASS_GAS,
+  DEVICE_CLASS_HEAT,
+  DEVICE_CLASS_LIGHT,
+  DEVICE_CLASS_LOCK,
+  DEVICE_CLASS_MOISTURE,
+  DEVICE_CLASS_MOTION,
+  DEVICE_CLASS_MOVING,
+  DEVICE_CLASS_OCCUPANCY,
+  DEVICE_CLASS_OPENING,
+  DEVICE_CLASS_PLUG,
+  DEVICE_CLASS_POWER,
+  DEVICE_CLASS_PRESENCE,
+  DEVICE_CLASS_PROBLEM,
+  DEVICE_CLASS_SAFETY,
+  DEVICE_CLASS_SMOKE,
+  DEVICE_CLASS_SOUND,
+  DEVICE_CLASS_VIBRATION,
+  DEVICE_CLASS_WINDOW,
+];
+
+export interface TemplateBinarySensor {
+  id: string;
+  name: string;
+  icon?: string;
+  device_class?: string;
+  value?: string;
+  availability?: string;
+  entity_picture?: string;
+  delay_on?: string;
+  delay_off?: string;
+}
+
+export interface TemplateBinarySensorMutableParams {
+  name: string;
+  icon?: string;
+  device_class?: string;
+  value?: string;
+  availability?: string;
+  entity_picture?: string;
+  delay_on?: string;
+  delay_off?: string;
+}
+
+export const fetchBinarySensor = (hass: HomeAssistant) =>
+  hass.callWS<TemplateBinarySensor[]>({ type: "binary_sensor/list" });
+
+export const createBinarySensor = (
+  hass: HomeAssistant,
+  values: TemplateBinarySensorMutableParams
+) =>
+  hass.callWS<TemplateBinarySensor>({
+    type: "binary_sensor/create",
```

Comment:
This type is wrong. It's creating a template binary sensor, not a binary sensor. So the type should represent this.

Reply:
And since it's part of the `template` integration, type should start with `template/`